src/main/java/com/limechain/storage/block/BlockState.java

```diff
@@ -904,4 +918,56 @@ public void handleFinalizedBlock(final Hash256 currentFinalizedHash) {
             //TODO: If currentFinalizedHash is not equal to subchain hash, delete subchain state trie
         }
     }
+
+    public synchronized void addBlockToBlockTree(BlockHeader blockHeader) {
+        if (isInitialized()) {
+            if (isFullSyncFinished()) {
+
+                processPendingBlocksFromQueue();
+
+                if (getPendingBlocksQueue().isEmpty()) {
+                    try {
+                        addBlock(new Block(blockHeader, new BlockBody(new ArrayList<>())));
+                    } catch (BlockStorageGenericException ex) {
+                        log.fine(String.format("[%s] %s", blockHeader.getHash().toString(), ex.getMessage()));
+                    }
+
+                    return;
+                }
+            }
+
+            addBlockToQueue(blockHeader);
+        }
+    }
```

Comment:
Imo it's good to avoid nested ifs when we can.

```suggestion
    public synchronized void addBlockToBlockTree(BlockHeader blockHeader) {
        if (!isInitialized()) {
            // either log and return or throw an exception
        }

        if (!isFullSyncFinished()) {
            addBlockToQueue(blockHeader);
            return;
        }
        
        processPendingBlocksFromQueue();

        if (getPendingBlocksQueue().isEmpty()) {
            try {
                addBlock(new Block(blockHeader, new BlockBody(new ArrayList<>())));
            } catch (BlockStorageGenericException ex) {
                log.fine(String.format("[%s] %s", blockHeader.getHash().toString(), ex.getMessage()));
            }
    }
```